java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -165,6 +225,48 @@ private CFG(BlockTree tree, Symbol.MethodSymbol symbol) {
     }
   }
 
+  private void prune() {
+    List<Block> inactiveBlocks = new ArrayList<>();
+    boolean first = true;
+    for (Block block : blocks) {
+      if (!first && isActive(block)) {
+        inactiveBlocks.add(block);
+      }
+      first = false;
+    }
+    if (!inactiveBlocks.isEmpty()) {
+      removeInactiveBlocks(inactiveBlocks);
+      if (inactiveBlocks.contains(currentBlock)) {
+        currentBlock = currentBlock.successors.iterator().next();
+      }
+      int id = 0;
+      for (Block block : blocks) {
+        block.id = id;
+        id += 1;
+      }
+    }
+  }
+
+  private boolean isActive(Block block) {
+    if (block == currentBlock) {
+      if (block.successors.size() > 1) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)